code/datums/elements/lazy_fishing_spot.dm

```diff
@@ -0,0 +1,25 @@
+// Lazy fishing spot element so fisheable turfs do not have a component each since they're usually pretty common on their respective maps (lava/water/etc)
+/datum/element/lazy_fishing_spot
+	element_flags = ELEMENT_BESPOKE | ELEMENT_DETACH
+	id_arg_index = 2
+	var/configuration
+
+/datum/element/lazy_fishing_spot/Attach(datum/target, configuration)
+	. = ..()
+	if(!isatom(target))
+		return ELEMENT_INCOMPATIBLE
+	if(!configuration)
+		CRASH("Lazy fishing spot had no configuration passed in.")
+	src.configuration = configuration
+
+	RegisterSignal(target, COMSIG_PRE_FISHING, .proc/create_fishing_spot)
+
+/datum/element/lazy_fishing_spot/Detach(datum/target)
+	. = ..()
+	UnregisterSignal(target, COMSIG_PRE_FISHING)
+
```

Comment:
when detaching or doing any sort of cleanup, it's usually the case that the parent gets called last